### ISSUE TITLE
Allowing tcp-ip nodes to join aws/azure-based cluster.

### ISSUE THREAD

**Issue by bjoernw** [hazelcast/hazelcast#3232]
I see that the AWS Joiner is a subclass of TcpIpJoiner and overrides the getType() method. That method is later used to determine whether a join request is of the same type that the existing cluster member used to join the cluster. But since the AWS Joiner is just a TcpIpJoiner shouldn't it be able to join? 

I'm running into this issue because I'm adding configuration for Azure and had to basically do everything you did for AWS. Now I have a cluster running in Azure and I'm trying to connect to it using an instance on my machine (I'm on the same virtual network as the cluster) and I'm getting a log message that i'm of a different joiner type, which I am since I'm using tcp-ip with a known set of private IP addresses.

This is the culprit: 

```
if (!joinerType.equals(other.joinerType)) {
    throw new HazelcastException("Incompatible joiners! " + joinerType + " -vs- " + other.joinerType);
}
```

Basically my question is whether this check is necessary and if it is whether I can just not override the getType() method in my Azure Joiner subclass to allow me to join using tcp-ip.

thanks!
